app/cdap/components/ConnectionList/Components/HeaderContent/index.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Box, IconButton } from '@material-ui/core';
+import SearchRounded from '@material-ui/icons/SearchRounded';
+import HeaderCustomTooltip from 'components/ConnectionList/Components/HeaderCustomTooltip';
+import HeaderSearch from 'components/ConnectionList/Components/HeaderSearch';
+import { useStyles } from 'components/ConnectionList/styles';
+import { IHeaderContentProps } from 'components/ConnectionList/types';
+import React from 'react';
+
+export default function({
+  eachFilteredData,
+  headersRefs,
+  index,
+  tabsData,
+  filteredData,
+  searchHandler,
+  makeCursorFocused,
+  handleSearch,
+  refs,
+  handleClearSearch,
+}: IHeaderContentProps) {
+  const classes = useStyles();
+
+  return (
+    <>
+      <Box
+        className={
+          eachFilteredData.toggleSearch
+            ? classes.hideComponent
+            : classes.beforeSearchIconClickDisplay
```

Comment:
can we refactor the classnames
i dont see this practise very good of using ternary ops here
plz try to make variations of a common component and use styled components inside the common component
just send props and get the required component with proper styles
i need this to be practised across the changes

Reply:
Done